### ISSUE TITLE
Reproduce questions related to the experimental data of the paper

### ISSUE THREAD

**Issue by liujj227** [zrguo/SuMi#2]
### Problem description
Hello author, after I downloaded the code, I only modified the batch_size=16 according to the instructions in the paper, and the other parameters were not changed. When reproducing a severity 5 **video corruption** experiment on the dataset **Kinetics50-C**, the accuracy of the SuMi method is two percentage points lower than that in the paper.
I wonder what went wrong, maybe you didn't use the checkpoints in the [READ ](https://github.com/XLearning-SCU/2024-ICLR-READ)paper? Or are there any hyperparameters that need to be modified?

### hyperparameters 
> uent_thresh=math.e/10, ment_thresh=math.log(1000)/2-1, umu=1.0, lamda=5.0, e_margin=math.log(1000)/2-1, batch_size=16

### Partial results
![Image](https://github.com/user-attachments/assets/b7f8d14c-13bd-48cc-bcd3-c02b507051ee)

**Comment by sarthaxxxxx** [comment]
I have similar issues as well, but you might wanna take a look at the hyperparameters used (as reported in the paper). The code uses the default values as you show. Tune it for Kinetics.

**Comment by zrguo** [comment]
Possible reasons may be:
- Differences between weak OOD, strong OOD and mixed domains. For example, we add mutual information loss for all iterations for weak OOD but only $t_0$ iterations for other setting. Besides, the parameter in [this line](https://github.com/zrguo/SuMi/blob/d0c08a855c4401e5b6a820bbd772852eaa78f40e/src/sumi.py#L69) varies across different configurations. We recommend referring to the paper for detailed hyperparameters.
- Dataset construction issue. Based on your reported results, some performances are better than those in the paper (e.g., glass blur), while others are lower (e.g., Gaussian). I noticed that Gaussian and shot performances are even below that of the source model in the paper. Have you tested the performance of the source model? In my experience, even with the same source model, performance can vary slightly (~1-2%) when using different datasets due to the random addition of noise. Therefore, I suggest testing the performance of the source model first. Moreover, it is advisable to focus on improvements over the source model and to compare results with other methods using the same source model and dataset.
- Regarding the Source Model: We utilize a pre-trained model and train it on the source domain to obtain the source model.

In addition to the above points, I recommend tuning your setup for better results, as differences in machines and software versions can impact performance.
